### PR TITLE
add docker support

### LINKED ISSUE #225: Docker image
Could we please have a docker image for this? It would make deployment much easier.

### PR DESCRIPTION
**Pull request by patrickdemers6**
# Description

Adding a `Dockerfile` to make it easier to run the tools provided. This will be most useful for those wishing to run the HTTP proxy in a production environment.

This introduces environment variable support for all flags available on the HTTP proxy. Environment variables are more graceful to use than command line flags when working with Docker.

I've done manual end to end testing, but would appreciate someone else setting this up themselves to help validate.

Fixes #225 

## Type of change

Please select all options that apply to this change:

- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update

# Checklist:

Confirm you have completed the following steps:

- [X] My code follows the style of this project.
- [X] I have performed a self-review of my code.
- [X] I have made corresponding updates to the documentation.
- [X] I have added/updated unit tests to cover my changes.

### CODE REVIEW

File: README.md
Comment (sethterashima) [inline README.md:58]: Other sections only capitalize the first word

File: README.md
Comment (sethterashima) [inline README.md:87]: Keep word wrapping consistent

File: README.md
Comment (sethterashima) [inline README.md:125]: Does the Fleet Telemetry server also allow configuration through environment variables? This could become confusing.

File: README.md
Comment (agbpatro) [inline README.md:89]: add docker pull command?

File: README.md
Comment (sethterashima) [inline README.md:130]: There are other timeouts used by the tools, so using a more specific name here could help avoid confusion over what timeout is being configured.

File: cmd/tesla-http-proxy/main.go
Comment (sethterashima) [inline cmd/tesla-http-proxy/main.go:43]: Is the root CA certificate required? They usually aren't presented by the server.

File: README.md
Comment (sethterashima) [inline README.md:98]: Make this a link

File: cmd/tesla-http-proxy/main.go
Comment (sethterashima) [inline cmd/tesla-http-proxy/main.go:157]: The environment variable should either allow the same time formatting options as Golang's `time.Duration` parser, or the environment variable should be renamed `TESLA_TIMEOUT_SECONDS` or similar.

File: cmd/tesla-http-proxy/main_test.go
Comment (sethterashima) [inline cmd/tesla-http-proxy/main_test.go:16]: Add `t.Helper()`

Comment (sethterashima) [reply cmd/tesla-http-proxy/main_test.go:16]: https://pkg.go.dev/testing#T.Helper

File: Dockerfile
Comment (sethterashima) [inline Dockerfile:8]: Add `config` to `.dockerignore`, along with `*.key` and `*.pem` files. Is `.git` automatically ignored?

File: cmd/tesla-http-proxy/main_test.go
Comment (sethterashima) [inline cmd/tesla-http-proxy/main_test.go:41]: Nit: other test names are lowercase

File: .github/workflows/publish.yml
Comment (agbpatro) [inline .github/workflows/publish.yml:34]: i think this repo uses this static file for version bump https://github.com/teslamotors/vehicle-command/blob/main/pkg/account/version.txt#L1
not sure if this will work as intended

Comment (sethterashima) [reply .github/workflows/publish.yml:34]: Yeah. I'm open to suggestions on how to manage this better, but the idea is to have client User Agent versions correspond to SDK versions. The Makefile updates it but `go build`/`go get` are more common than `make`.

File: README.md
Comment (patrickdemers6) [reply README.md:125]: It's generally configured with `config.json` instead of flags/env vars.

File: cmd/tesla-http-proxy/main.go
Comment (patrickdemers6) [reply cmd/tesla-http-proxy/main.go:43]: It should be included if the cert is signed by a CA. Running the HTTP proxy without a cert specified fails.

This PR isn't touching the help message but happy to tweak if this is not accurate.

```go
func http.ListenAndServeTLS(addr string, certFile string, keyFile string, handler http.Handler) error
// ListenAndServeTLS acts identically to ListenAndServe,
// except that it expects HTTPS connections. Additionally, files containing a
// certificate and matching private key for the server must be provided.
// If the certificate is signed by a certificate authority, the certFile should
// be the concatenation of the server's certificate, any intermediates, and the CA's certificate.
```

File: README.md
Comment (patrickdemers6) [reply README.md:92]: Keeping docker run, but switching it to just printing usage. Not diving into all the args since they haven't been explained yet.

File: .github/workflows/publish.yml
Comment (patrickdemers6) [reply .github/workflows/publish.yml:34]: Revised `publish.yml` so it automatically creates a GitHub release and pushes to Docker hub when the `version.txt` file changes. 

Open to suggestions on how to test the workflow before merging.

Comment (patrickdemers6) [reply .github/workflows/publish.yml]: Just tested this in a private repository and everything is working well.

GitHub release is auto created and the image is pushed to Docker with `latest` tag and version tag. When something is merged and version tag doesn't change, nothing is triggered.

Comment (patrickdemers6) [reply .github/workflows/publish.yml]: We will need to make sure the Docker secrets are set and the workflow runner has write access to the repository.

**Review by agbpatro — APPROVED** [review]

File: README.md
Comment (sethterashima) [inline README.md:249]: This exposes 4433 on the host, right? Let's make the example command use `-p 127.0.0.1:4433:4433` so people who are unfamiliar with docker don't inadvertently expose the proxy. We can add directions to omit the localhost IP if exposing the service is intentional.

File: .dockerignore
Comment (sethterashima) [inline .dockerignore:2]: Some others we should add: `*.jwt`, `*.jws`.

**Review by sethterashima — APPROVED** [review]

**Pull request merged**
